pygmt/figure.py

```diff
@@ -236,13 +236,17 @@ def savefig(
 
         prefix, ext = os.path.splitext(fname)
         ext = ext[1:]  # Remove the .
+        if ext == "ps":
+            raise GMTInvalidInput(
+                "Extension '.ps' is not supported. Please use '.eps' or '.pdf' instead."
+            )
         if ext not in fmts:
-            raise GMTInvalidInput("Unknown extension '.{}'".format(ext))
+            raise GMTInvalidInput(f"Unknown extension '.{ext}'.")
```

Comment:
Maybe put the new if-block here after `if ext not in fmts`?

```suggestion
            if ext == "ps":
                raise GMTInvalidInput(
                    "Extension '.ps' is not supported. "
                    "Please use '.eps' or '.pdf' instead."
                )
            raise GMTInvalidInput(f"Unknown extension '.{ext}'.")
```